src/libstd/sync/once.rs

```diff
@@ -323,7 +323,7 @@ impl Once {
                 POISONED |
                 INCOMPLETE => {
                     let old = self.state.compare_and_swap(state, RUNNING,
-                                                          Ordering::SeqCst);
+                                                          Ordering::AcqRel);
```

Comment:
These could use `compare_exchange` and a `Ordering::Relaxed` in the `failure` case, I believe. Do you have a way to verify/check that?

Reply:
I could try to bribe Hai or @jhjourdan into proving it. ;) But that's likely not going to happen soon.